js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+/* global console */
+/* global tinyMCE */
+/* global wpseoShortcodePluginL10n */
+/* global ajaxurl */
+/* global YoastSEO */
+/* global yoastShortcodePlugin */
+/* jshint -W097 */
+/* jshint -W003 */
+'use strict';
+
+/**
+ * The Yoast Shortcode plugin parses the shortcodes in a given piece of text. It analyzes multiple input fields for shortcodes which it will preload using AJAX.
+ *
+ * @constructor
+ */
+var YoastShortcodePlugin = function() {
+	this.registerTime = 0;
+	this.register( this.bindElementEvents );
+
+	this.unparsedShortcodes = [];
+	this.parsedShortcodes = [];
+
+	// The regex for matching shortcodes based on the available shortcode keywords.
+	this.keywordRegex = '(' + wpseoShortcodePluginL10n.wpseo_shortcode_tags.join('|') + ')';
+
+	this.loadShortcodes();
+};
+
+/* YOAST SEO CLIENT */
+
+/**
+ * Register the plugin with YoastSEO, retry for one second.
+ *
+ * @param {function} callback (optional) function to call when the plugin has been registered. This is useful for event bindings which rely on YoastSEO to be there.
+ */
+YoastShortcodePlugin.prototype.register = function( callback ) {
+	if ( typeof YoastSEO !== 'undefined' && typeof YoastSEO.app !== 'undefined' && typeof YoastSEO.app.plugins !== 'undefined' ) {
+		YoastSEO.app.plugins.register( 'YoastShortcodePlugin', { status: 'loading' } );
+		if ( typeof callback === 'function' ) {
+			callback();
+		}
+	}
+	else if ( this.registerTime < 1001 ) {
```

Comment:
This is bound to go wrong with slow servers or slow browsers. We should just make sure we include this file after including `YoastSEO`. YoastSEO is a hard dependency anyway so we should define this in our PHP as well.